<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Chip8 Emulator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QFrame" name="displayFrame">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>640</width>
        <height>320</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="displayLayout"/>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="controlsLayout">
      <item>
       <widget class="QPushButton" name="resetButton">
        <property name="text">
         <string>Reset</string>
        </property>
        <property name="toolTip">
         <string>Reset the emulator</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pauseButton">
        <property name="text">
         <string>Pause</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="speedLabel">
        <property name="text">
         <string>Speed:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="speedSlider">
        <property name="minimum">
         <number>100</number>
        </property>
        <property name="maximum">
         <number>2000</number>
        </property>
        <property name="value">
         <number>700</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksBelow</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="speedValueLabel">
        <property name="minimumSize">
         <size>
          <width>50</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>700 Hz</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_ROM"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEmulation">
    <property name="title">
     <string>Emulation</string>
    </property>
    <addaction name="actionPause"/>
    <addaction name="actionReset"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionKeyboard_Mapping"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="actionDisplay_Settings"/>
    <addaction name="actionQuirks"/>
    <addaction name="actionLanguage_Settings"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <widget class="QMenu" name="menuTheme">
     <property name="title">
      <string>Theme</string>
     </property>
     <property name="icon">
      <iconset theme="preferences-desktop-theme"/>
     </property>
     <addaction name="actionLightTheme"/>
     <addaction name="actionDarkTheme"/>
     <addaction name="separator"/>
     <addaction name="actionSystemTheme"/>
    </widget>
    <addaction name="menuTheme"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEmulation"/>
   <addaction name="menuView"/>
   <addaction name="menuSettings"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionLoad_ROM">
   <property name="icon">
    <iconset theme="document-open"/>
   </property>
   <property name="text">
    <string>Load ROM...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset theme="application-exit"/>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionPause">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="media-playback-pause"/>
   </property>
   <property name="text">
    <string>Pause</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="icon">
    <iconset theme="view-refresh"/>
   </property>
   <property name="text">
    <string>Reset</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="icon">
    <iconset theme="preferences-system"/>
   </property>
   <property name="text">
    <string>Settings...</string>
   </property>
  </action>
  <action name="actionDisplay_Settings">
   <property name="icon">
    <iconset theme="preferences-desktop-display"/>
   </property>
   <property name="text">
    <string>Display Colors...</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset theme="help-about"/>
   </property>
   <property name="text">
    <string>About...</string>
   </property>
  </action>
  <action name="actionKeyboard_Mapping">
   <property name="icon">
    <iconset theme="input-keyboard"/>
   </property>
   <property name="text">
    <string>Keyboard Mapping</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+K</string>
   </property>
  </action>
  <action name="actionQuirks">
   <property name="icon">
    <iconset theme="preferences-system"/>
   </property>
   <property name="text">
    <string>Quirks Configuration...</string>
   </property>
  </action>
  <action name="actionLanguage_Settings">
   <property name="icon">
    <iconset theme="preferences-desktop-locale"/>
   </property>
   <property name="text">
    <string>Language...</string>
   </property>
  </action>
  <action name="actionLightTheme">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="weather-clear"/>
   </property>
   <property name="text">
    <string>Light Mode</string>
   </property>
  </action>
  <action name="actionDarkTheme">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="weather-clear-night"/>
   </property>
   <property name="text">
    <string>Dark Mode</string>
   </property>
  </action>
  <action name="actionSystemTheme">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="computer"/>
   </property>
   <property name="text">
    <string>Use System Theme</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
